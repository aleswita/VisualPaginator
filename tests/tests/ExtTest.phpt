<?php

/**
 * This file is part of the AlesWita\Components\VisualPaginator
 * Copyright (c) 2015 Ales Wita (aleswita+github@gmail.com)
 *
 * @phpVersion 7.1.0
 */

declare(strict_types=1);

namespace AlesWita\Components\VisualPaginator\Tests\Tests;

use AlesWita;
use Nette;
use Tester;

require_once __DIR__ . "/../bootstrap.php";


final class ExtTest extends Tester\TestCase
{
	/**
	 * @return void
	 */
	public function setUp(): void {
		parent::setUp();
	}

	/**
	 * @return void
	 */
	public function tearDown(): void {
		parent::tearDown();
	}

	/**
	 * @param string
	 * @return Nette\DI\Container
	 */
	private function createContainer(string $config): Nette\DI\Container {
		$configurator = new Nette\Configurator();

		$configurator->setTempDirectory(TEMP_DIR);
		$configurator->addConfig($config);

		return $configurator->createContainer();
	}

	/**
	 * @return Nette\Application\IPresenter
	 */
	private function createPresenter(): Nette\Application\IPresenter {
		$presenter = $this->presenterFactory->createPresenter("Test");
		$presenter->autoCanonicalize = FALSE;
		return $presenter;
	}

	/**
	 * @throws Nette\InvalidArgumentException
	 * @return void
	 */
	public function testOne(): void {
		$configurator = new Nette\Configurator();

		$configurator->setTempDirectory(TEMP_DIR);
		$configurator->addConfig(__DIR__ . "/../app/config/extTest1.neon");

		$configurator->createContainer();
	}

	/**
	 * @throws Nette\InvalidArgumentException
	 * @return void
	 */
	public function testTwo(): void {
		$configurator = new Nette\Configurator();

		$configurator->setTempDirectory(TEMP_DIR);
		$configurator->addConfig(__DIR__ . "/../app/config/extTest2.neon");

		$configurator->createContainer();
	}

	/**
	 * @throws Nette\InvalidArgumentException
	 * @return void
	 */
	public function testThree(): void {
		$configurator = new Nette\Configurator();

		$configurator->setTempDirectory(TEMP_DIR);
		$configurator->addConfig(__DIR__ . "/../app/config/extTest3.neon");

		$configurator->createContainer();
	}

	/**
	 * @throws Nette\InvalidArgumentException
	 * @return void
	 */
	public function testFour(): void {
		$configurator = new Nette\Configurator();

		$configurator->setTempDirectory(TEMP_DIR);
		$configurator->addConfig(__DIR__ . "/../app/config/extTest4.neon");

		$configurator->createContainer();
	}

	/**
	 * @throws Nette\InvalidArgumentException
	 * @return void
	 */
	public function testFive(): void {
		$configurator = new Nette\Configurator();

		$configurator->setTempDirectory(TEMP_DIR);
		$configurator->addConfig(__DIR__ . "/../app/config/extTest5.neon");

		$configurator->createContainer();
	}

	/**
	 * @throws Nette\InvalidArgumentException
	 * @return void
	 */
	public function testSix(): void {
		$configurator = new Nette\Configurator();

		$configurator->setTempDirectory(TEMP_DIR);
		$configurator->addConfig(__DIR__ . "/../app/config/extTest6.neon");

		$configurator->createContainer();
	}
}


$test = new ExtTest;
$test->run();
