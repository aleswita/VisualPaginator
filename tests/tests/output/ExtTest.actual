array(
	'main' => 'C:\\xampp\\htdocs\\easyweb\\vendor\\aleswita\\visualpaginator\\src/templates/normal/main.latte',
	'paginator' => 'C:\\xampp\\htdocs\\easyweb\\vendor\\aleswita\\visualpaginator\\src/templates/normal/paginator.latte',
	'itemsPerPage' => 'C:\\xampp\\htdocs\\easyweb\\vendor\\aleswita\\visualpaginator\\src/templates/normal/items-per-page.latte',
)
