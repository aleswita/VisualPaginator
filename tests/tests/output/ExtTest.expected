array(
	'main' => 'C:\\xampp\\htdocs\\easyweb\\vendor\\aleswita\\visualpaginator\\src/templates/bootstrap-v4/main.latte',
	'paginator' => 'C:\\xampp\\htdocs\\easyweb\\vendor\\aleswita\\visualpaginator\\src/templates/bootstrap-v4/paginator.latte',
	'itemsPerPage' => 'C:\\xampp\\htdocs\\easyweb\\vendor\\aleswita\\visualpaginator\\src/templates/bootstrap-v4/items-per-page.latte',
)
